packages/schema-generator/src/generate_schema.ts

```diff
@@ -0,0 +1,100 @@
+import { TypescriptDataSchemaGenerator } from '@aws-amplify/graphql-schema-generator';
+import * as fs from 'fs';
+import { AmplifyUserError } from '@aws-amplify/platform-core';
+
+export type SchemaGeneratorConfig = {
+  connectionString: string;
+  out: string;
+};
+
+/**
+ * Schema generator class.
+ */
+export class SchemaGenerator {
+  generate = async (props: SchemaGeneratorConfig) => {
+    const dbConfig = parseDatabaseUrl(props.connectionString);
+
+    try {
+      const schema = await TypescriptDataSchemaGenerator.generate(dbConfig);
+      fs.writeFileSync(props.out, schema);
+    } catch (err) {
+      const databaseError = err as DatabaseConnectError;
+      if (databaseError.code === 'ETIMEDOUT') {
+        throw new AmplifyUserError(
+          'DatabaseConnectionError',
+          {
+            message: `Unable to connect to the database at ${dbConfig.host}:${dbConfig.port}. `,
+            resolution:
+              'Check if the credentials are correct. Also, check if the database is running and accessible from the network.',
+          },
+          databaseError
+        );
+      }
+      throw err;
+    }
+  };
+}
+
+// eslint-disable-next-line spellcheck/spell-checker
+export type MySQLEngine = 'mysql';
+// eslint-disable-next-line spellcheck/spell-checker
+export type PostgresSQLEngine = 'postgresql';
+// eslint-disable-next-line spellcheck/spell-checker
+export type SQLEngine = MySQLEngine | PostgresSQLEngine;
+
+export type SQLDataSourceConfig = {
+  engine: SQLEngine;
+  username: string;
+  password: string;
+  host: string;
+  database: string;
+  port: number;
+};
+
+/**
+ * Error for database connection failures.
+ */
+export class DatabaseConnectError extends Error {
+  /**
+   * Creates database connection error.
+   */
+  // eslint-disable-next-line spellcheck/spell-checker
```

Comment:
can't we rename `errorno` to `errorNo`?

Reply:
Please drop a comment next to suppression explaining that then.

Reply:
I'm still not seeing why we can't change the prop name? The ctor signature is still the same, just rename `errorno` to `errorNo`.

Reply:
@edwardfoyle I had this doubt today as well.

after few minutes of staring at this i noticed `readonly` keyword which makes it class prop (which presumably is part of that super type).

Reply:
which also means that this could be refactored into :
make parameter `errorNo` and then in ctor call `this.errorno = errorNo` - @sundersc maybe we should do that instead suppression ?

Reply:
Added a comment.

Reply:
I'm not sure I follow. I do not create an instance of this type anywhere directly, I'm using it to cast an error thrown by the underlying knex library.

Reply:
Actually errorno is not used, so I have removed it. Also marking the class as abstract as we don't intend to create instance of this class.